### ISSUE TITLE
"Something went wrong. Please try again"

### ISSUE THREAD

**Issue by Fiiily** [Lavoaster/IP.Board-Steam-Authentication-Method#87]
on IPS cloud installation. see my forums @ forums.juicylandgaming.com to see the error. 
it sometimes works, it sometimes doesn't. let me know if there are any logs that you need.

**Comment by madman32395** [comment]
should have a log printed for it in System Error or Error Logs... and what action is being taken to get this error to occur? logging in? registering? acp login?

**Comment by madman32395** [comment]
I was able to register with steam just fine without error, logged out and back in without issue as well.

___________
also whats the app you use to prompt steam only login?

**Comment by Hyrius** [comment]
I'm having the same issue. 
Please note that this is using IPB 4.2.0.

**Comment by Lavoaster** [comment]
At some point this week I'm going to be performing an upgrade to 4.2.0 on a site I help maintain, I'll look into at that time.

**Comment by Hyrius** [comment]
I just updated my IPB to 4.2.1, and it seems to be working again. 
Can you confirm, @Fiiily?

**Comment by MStruntze** [comment]
@Hyrius still seems to be an issue now and then. Fresh install with 4.2.1 and this plugin errors quite a lot (something went wrong) when attempting to register.
No errors seem to be generated in logs.

**Comment by Hyrius** [comment]
@MStruntze Agreed. I get errors sometimes, too.

**Comment by Hyrius** [comment]
@Lavoaster Sorry to disturb, but... Any news on this one? :)

**Comment by Fiiily** [comment]
@madman32395 Hey, how's it going? I'm sorry for getting back to you late.
The issue is resolved with recent IPS updates and I am using template modifications to lock you into using Steam authentication.

**Comment by robinkooli** [comment]
Same issue.

**Comment by seboxxl** [comment]
I've got this error every time

**Comment by Phil7789** [comment]
I can confirm this, no sign-in possible.
The plugin pulls the profile picture from steam but signing in is not possible. There are no error logs written. I cound this as **critical**
IPB 4.2.5

**Comment by sannysc** [comment]
Also having this issue.

**Comment by Fiiily** [comment]
I don't think IPS Cloud's www server is misconfig'd.

**Comment by jrod9k1** [comment]
Having the same issue here. Not super familiar with PHP however through a little editing I've identified it's throwing the exception found here: https://github.com/Lavoaster/IP.Board-Steam-Authentication-Method/blob/master/upload/system/Login/Steam.php#L77

I hope that helps. I'm gonna attempt to fumble my way further through this and hopefully come back with a little more info and maybe even a pull request if I get lucky.

**Comment by seboxxl** [comment]
Like others said - it's randomly working (or not). Sometimes I need to try logging 5 times before I log in.
@jrod9k1 the problem is with private function validate(), I guess.

**Comment by Phil7789** [comment]
I did some test on the 2FA with the Steam App, perhaps other users could try this too.

I'm using the App with the Code generator and have those said problems randomly as the login is on halt until I enter my 2FA code. A buddy of mine does not use the 2FA and could log in instantly without any problems.

**Comment by WansonSwanson** [comment]
I seem to be having the same issue on 4.2.6. Logins work at random, sometimes they work the first time, other times it takes 3-4 tries.

**Comment by WansonSwanson** [comment]
I was able to solve my issue by disabling force friendly urls in the seo settings. Seems to be an issue with this setting.

**Comment by Hawk-HQ** [comment]
I have the exact problem, sometimes it works, sometimes it doesn't. This is a big problem trying to grow a community. Please fix this

**Comment by Hyrius** [comment]
@madman32395 Any news?

**Comment by Lebofly** [comment]
I can confirm that this was solved for me by turning off friendly URLS

**Comment by madman32395** [comment]
for giggles, try again after updating against #98  ; another community that I can play with, this seemed to fix "Something went wrong. Please try again".

**Comment by madman32395** [comment]
awesome. let us know what happens :)

**Comment by Lebofly** [comment]
So far so good, but will have to wait and see because it only occurred occasionally for myself

**Comment by JMan49** [comment]
Was working fine on IPB 4.2.8. Now getting "Something went wrong. Please try again." Tried upgrading to Sign in through Steam - 2.0.4 still same issue. Even tried uninstalling to no avail. This is a cloud IPS community. Nothing I have tried has fixed this.

**Comment by madman32395** [comment]
@JMan49 for a CiC setup have you had ips upload the files in upload folder?

**Comment by JMan49** [comment]
@madman32395 Yeah as far as I know they have.

**Comment by Lavoaster** [comment]
As I'll be working on this soon for the 4.3 upgrade, did anyone actually come up with a reproducible way to produce this?

**Comment by Lavoaster** [comment]
Please download [this](https://github.com/Lavoaster/IPS-SteamLoginMethod/releases/download/v3.0.0/steamlogin-3.0.0.zip) and follow the instructions in the readme for upgrading.

The update will either resolve your issue, or make it more obvious what the error is now as I've all errors that I throw in the login method much more verbose, and less generic. E.G Unable to connect to Steam, Unable to validate OpenID credentials, etc...

Also I found out during the that it was possible for a steam account to be linked more than one forum account, which would then show this error message. During the update it will remove all duplicates and force them to resync again. Thanks to Aiwa for this and for the code that helps sort it out!

Let me know if you have any issues, this should be up on the marketplace soon.

**Comment by jrod9k1** [comment]
@Lavoaster 

Just tested the latest version per the instructions on readme.md. Working fantastic! I did notice one minor issue that I inadvertently caused:

After I installed the plugin I ended up deleting and re-adding the steam login handler for circumstantial reasons. That seemed to iterate the "login_id" of the handler from 10 to 11 under "core_login_methods". After doing so the "token_login_method" under "core_login_links" was still set to 10 per the old method causing existing steam users to be prompted to set up a new account. Not sure if it's practical but perhaps have some way to detect if the handler was deleted/re-added and auto-update "core_login_links"

Thanks for all of your hard work on this!

**Comment by itsAliAvali** [comment]
If the linked auths are still there but references the uninstalled handler
you may have to run a manual query to change them from 10 to 11 (or what
ever number the steam login is) , so they can login normally.

**Comment by jrod9k1** [comment]
@madman32395 Yeah I've already updated the links to the new handler, that wasn't an issue. Was more just noting that I caused the problem to begin with as other people using the plugin may not be technically apt enough to fix it on their own.
